### PR TITLE
apps: testing/ostest: Add comprehensive scheduler parameter tests

### PR DESCRIPTION
**Pull request by hujun260**
This PR introduces a comprehensive test suite for NuttX scheduler parameter functions, implemented as part of the ostest application framework.

### Summary
This contribution adds a new scheduler parameter testing module to the ostest application. The test suite validates the core scheduler API functions that control task scheduling priority and policy, ensuring their correct operation across various scenarios and edge cases.

### TEST

./tools/configure.sh qemu-armv7r:nsh

```
user_main: scheduler parameters test

sched_params_test: Starting scheduler parameters test

Test 1: Get main task scheduler and priority
sched_params_test: Main task PID = 4
sched_params_test: Main task policy = 2 (SCHED_FIFO=1, SCHED_RR=2)
sched_params_test: Main task priority = 100

Test 2: Set priority and verify
sched_params_test: Set priority to 150
sched_params_test: Verified priority set to 150

Test 3: Test thread parameter modification
sched_params_test: Test thread created
sched_params_test: Test thread initial priority = 100

Test 4: Test invalid PID handling
sched_params_test: Correctly rejected invalid PID

Test 5: Get scheduler for self (PID 0)
sched_params_test: Self scheduler policy = 2

Test 6: Get parameters for self (PID 0)
sched_params_test: Self priority = 150

Test 7: Multiple priority changes
sched_params_test: [1] Priority changed to 120 and verified
sched_params_test: [2] Priority changed to 140 and verified
sched_params_test: [3] Priority changed to 160 and verified
sched_params_test: [4] Priority changed to 130 and verified

Test 8: Restored original priority to 100

sched_params_test: Canceling test thread
sched_params_test: Test thread cleaned up

sched_params_test: All scheduler parameter tests PASSED
```

### CODE REVIEW

File: testing/ostest/sched_params.c [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+  /* Test 1: Get main task's current scheduler and parameters */
+
+  printf("\nTest 1: Get main task scheduler and priority\n");
+
+  main_pid = getpid();
+  printf("sched_params_test: Main task PID = %d\n", main_pid);
+
+  /* Get scheduler policy for main task */
+
+  ret = nxsched_get_scheduler(main_pid);
```
Comment (xiaoxiang781216) [inline testing/ostest/sched_params.c:132]: why not call sched_get_xxx, so protect/kernel build can run this test case?
